src/lib.rs

```diff
@@ -814,6 +814,12 @@ impl<K: EnrKey> Enr<K> {
         };
         Ok(())
     }
+
+    /// Consumes the type and returns all pairs of ENR
+    #[must_use]
+    pub fn into_pairs(self) -> BTreeMap<Key, Bytes> {
+        self.content
+    }
```

Comment:
returning the `BTreeMap` here ties an implementation detail (the use of a `BTreeMap`) to a public API. I suggest you implement `IntoIterator` for `Enr<K>` instead

Reply:
these are the k,v pairs of the ENR, which are sorted and unique, imo returning `BTreeMap` is fitting here.

In order to get all pairs as a map I'd need to do `into_iter().collect::<BTreeMap<_,_>()`.

But maybe `IntoIterator` is just more appropriate given that there's already `Iter`?

Reply:
Returning an iterator /implementing `IntoIter` will produce the pairs in sorted order and unique simply because it's part of the spec. There are valid implementations of an ENR that could use something beside a `BTreeMap` and thus is why making it explicit would not be a best practice. 

This is also why, as you mention, the `iter` function does also avoid exposing implementation details and simply declares a [`impl Iterator`](https://github.com/sigp/enr/blob/8386e3cac279a81b76368fd5e2845a4dc4c875c4/src/lib.rs#L274)